flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/table.scala

```diff
@@ -150,7 +150,35 @@ class Table(
     * }}}
     */
   def filter(predicate: Expression): Table = {
-    new Table(tableEnv, Filter(predicate, logicalPlan).validate(tableEnv))
+
+    predicate match {
```

Comment:
The IN operator can also be used in a `select()` statement but if we do the translation here, only `filter()` is supported.